packages/components/Button/src/useButton.ts

```diff
@@ -7,13 +7,13 @@ import { AccessibilityState } from 'react-native';
 export const useButton = (props: ButtonProps): ButtonState => {
   // attach the pressable state handlers
   const defaultComponentRef = React.useRef(null);
-  const { onClick, accessibilityState, componentRef = defaultComponentRef, disabled, loading, ...rest } = props;
+  const { onClick, accessibilityState, componentRef = defaultComponentRef, disabled, loading } = props;
+  const isDisabled = !!disabled || !!loading;
   // GH #1336: Set focusRef to null if button is disabled to prevent getting keyboard focus.
-  const focusRef = disabled ? null : componentRef;
+  const focusRef = isDisabled ? null : componentRef;
   const onClickWithFocus = useOnPressWithFocus(focusRef, onClick);
-  const pressable = useAsPressable({ ...rest, onPress: onClickWithFocus });
+  const pressable = useAsPressable({ ...props, onPress: onClickWithFocus });
```

Comment:
I think you'll need to specifically override disabled and pass disabled: isDisabled here

Reply:
Ah right, good catch.